java/src/jmri/BlockManager.java

```diff
@@ -42,6 +44,9 @@ public BlockManager() {
         super();
         InstanceManager.sensorManagerInstance().addVetoableChangeListener(this);
         InstanceManager.getDefault(ReporterManager.class).addVetoableChangeListener(this);
+        InstanceManager.getList(PowerManager.class).forEach((pm) -> {
+            pm.addPropertyChangeListener(this);
+        });
```

Comment:
You also need to listen to the InstanceManager for changes in the property returned by `InstanceManager.getListPropertyName(PowerManager.class)` so you can listen to new PowerManagers (should the Block manager be created before a PowerManager, and remove the BlockManager as a listener from removed PowerManagers.